### PR TITLE
Forbid reserved keywords as usernames

### PR DESCRIPTION
**Pull request by afzalsayed96**
## What does this PR do?

Prevents registering key username such as `jobs`

Screenshot 2022-08-25 at 8 19 53 PM

image

### CODE REVIEW

File: packages/features/ee/common/lib/checkPremiumUsername.ts [hunk 1]
```diff
@@ -10,6 +10,14 @@ interface ResponseUsernameApi {
 
 export async function checkPremiumUsername(_username: string): Promise<ResponseUsernameApi> {
   const username = slugify(_username);
+  if (username in FORBIDDEN_USERNAMES) {
```
Comment (afzalsayed96) [inline packages/features/ee/common/lib/checkPremiumUsername.ts:13]: Assuming `key in object` will be faster than `array.includes(string)`

File: packages/features/ee/common/lib/checkPremiumUsername.ts [hunk 2]
```diff
@@ -10,6 +10,14 @@ interface ResponseUsernameApi {
...
+    return {
+      available: false as const,
+      premium: false,
+      message: "A user exists with that username",
```
Comment (afzalsayed96) [inline packages/features/ee/common/lib/checkPremiumUsername.ts:17]: This message can be changed

File: packages/lib/constants.ts [hunk 3]
```diff
@@ -36,3 +36,14 @@ export const DEVELOPER_DOCS = "https://developer.cal.com";
...
+  about: "about",
+  blog: "blog",
+  download: "download",
+  jobs: "jobs",
+  open: "open",
+  privacy: "privacy",
+  roadmap: "roadmap",
+  security: "security",
+  terms: "terms",
+};
```
Comment (afzalsayed96) [inline packages/lib/constants.ts:49]: Not exhaustive but covers most clashes with current website pages

**Comment by PeerRich** [comment]
cool idea! luckily those accounts dont affect website behaviour but its probably a good idea

**Comment by PeerRich** [comment]
alternatively we can just create accouts for each page that we create. not clean but also gets the job done

**Review by zomars — CHANGES_REQUESTED** [review]
Thank you for your contribution 🙏 

- I would rather have this at a per-database level since self-hosted shouldn't have to worry about this.
- We already have some guards on our private website repo
- I'd rather have as less hardcoding as possible

**Pull request closed without merging**